db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/DB2LengthOperatorNode.java

```diff
@@ -194,5 +194,17 @@ private int getConstantNodeLength() throws StandardException
         if( operand instanceof ConstantNode)
             return ((ConstantNode) operand).getValue().getLength();
         return -1;
-    }        
+    }
+
+    @Override
+    public double getBaseOperationCost() throws StandardException {
+        double cost = 0.0;
+        if (operand != null) {
+            if (getConstantLength() == -1) {
+                cost += SIMPLE_OP_COST * operand.getTypeServices().getNull().getLength();
```

Comment:
Is the cost really multiplied by the length of the data type?  What if you have a really long string, is it really that expensive?

Reply:
I have the same concern here actually...I took a look at `SQLChar.getLength()` and it seems to have chance to calculate it in O(n). But then the expression evaluation cost could accumulate fast, soon dominating the plan cost. In the latest commit, I add an upper bound on this, hopefully better than linear.